### PR TITLE
Subviews

### PR DESCRIPTION
**Pull request by Kingdorian**
I changed so now preset creation is based on subviews (which can later be used for auto preset generation tweaking in the ui). They'll also be used to draw rectangles in the preview so the user can see where the automatically created presets will be.

@bvanwijk could you take a look at the failing test for serverController?

### CODE REVIEW

File: backend/src/main/java/com/benine/backend/preset/IPCameraPreset.java [hunk 1]
```diff
@@ -65,7 +64,7 @@ public IPCameraPreset(IPCamera cam, int panSpeed, int tiltSpeed)
     this.position = new ZoomPosition(cam.getPosition(), cam.getZoom());
     this.focus = cam.getFocusPosition();
     this.iris = cam.getIrisPosition();
-    this.autofocus = cam.isAutoFocusOn();
+    this.autoiris = cam.isAutoFocusOn();
```
Comment (bvanwijk) [inline backend/src/main/java/com/benine/backend/preset/IPCameraPreset.java]: I think this should be this.autoriris = cam.isAutoIrisOn()

**Comment by bvanwijk** [comment]
Except from the comment I made, it looks great.

File: backend/src/main/java/com/benine/backend/preset/IPCameraPreset.java (hunk 1 above)
Comment (Kingdorian) [inline backend/src/main/java/com/benine/backend/preset/IPCameraPreset.java]: Great find I switched those two around I'll fix it.

**Comment by jochem725** [comment]
Looks like a great feature! I think this can be merged.

**Comment by egedebruin** [comment]
Looks really good, I also think this can be merged.

**Pull request merged**